docs/my_documentation.rst

```diff
@@ -0,0 +1,365 @@
+
+.. code-block:: python
+
+	@multipass.identity_handler
+	def process_identity(identity_info):
+	    identity = Identity.query.filter_by(provider=identity_info.provider.name,
+		                                identifier=identity_info.identifier).first()
+	    if identity is None:
+			user = User.query.filter_by(email=identity_info.data['email']).first()
+			if not user:
+		    	# Redirect to register page
+		    	return redirect(url_for('auth.register', provider=identity_info.provider.name))
+			else:
+			    # Create new identity and link with already existing account
+			    identity = Identity(user=user, provider=identity_info.provider.name,
+						identifier=identity_info.identifier, data=identity_info['data'],
+						multipass_data=identity_info['multipass_data'])
+			    user.identities.append(identity)
+	    else:
+			user = identity.user
+	    login_user(user, identity)
+
+If ``multipass.identity_handler`` decorated method doesn't return anything, 
+``handle_auth_success`` returns ``redirect_success`` which redirects to URL stored in ``session`` as ``'_multipass_next_url'``. This URL can be set using the ``set_next_url`` method :
+
+.. code-block:: python
+
+    def set_next_url(self):
+        """Saves the URL to redirect to after logging in."""
+        next_url = request.args.get('next')
+        if next_url:
+            session['_multipass_next_url'] = next_url
+
+If the ``'_multipass_next_url'`` is not set, ``redirect_success`` redirects to ``MULTIPASS_SUCCESS_ENDPOINT`` from Multipass configuration.
+
+
+Failed authentication
+---------------------------
+
+In case the autherntication was unsuccessful, and ``MultipassException`` was raised, ``handle_auth_error`` flashes the ``MULTIPASS_FAILURE_MESSAGE`` and if ``recdirect_to_login`` argument is set, it redirects to ``MULTIPASS_LOGIN_ENDPOINT``
+
+
+Logout
+---------
+
+``multipass.logout`` should be called by your application upon logout request, passing it the url to redirect to after logout and optionaly a flag to clear session.
+
+.. code-block:: python
+
+	return multipass.logout(request.args.get('next'), clear_session=True)
+
+The ``logout`` method then calls  ``process_logout`` on provider which name was stored in ``session`` as ``'_multipass_login_provider'`` upon login. In ``process_logout``method the provider can implement some provider-specific actions such as sending a logout notification to the provider or redirecting to a SSO logout page. The ``return_url`` from argument can be passed further if the external provider allows to specify the URL to redirect to after logging out.
```

Comment:
The absence of an space after `process_logout` is messing things up in the sentence.